core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -502,6 +503,21 @@ class KafkaController(val config: KafkaConfig,
     info("Resigned")
   }
 
+  private def removeMetrics(): Unit = {
+    metricsGroup.removeMetric("ActiveControllerCount")
```

Comment:
Perhaps we should store the metric names in an Array and then close all of them with:

```
for (String metricName : metrics)
                metricsGroup.removeMetric(metricName);
```

this will ensure that new metrics added in the future here are cleaned up.